code/obj/mining_cloud_storage.dm

```diff
@@ -275,24 +263,30 @@
 			OCD.price = max(0,new_price)
 		return
 
-	proc/eject_ores(var/ore, var/eject_location, var/ejectamt, var/transmit = 0, var/user as mob)
+	proc/eject_ores(var/material_name, var/eject_location, var/ejectamt, var/transmit = FALSE, var/mob/user)
+		if(ejectamt < 1)
+			return
 		var/amount_ejected = 0
 		if(!eject_location)
 			eject_location = get_output_location()
-		for(var/obj/item/raw_material/R in src.contents)
-			if (R.material_name == ore)
-				R.set_loc(eject_location)
-				ejectamt--
-				amount_ejected++
-				if (ejectamt <= 0)
-					break
+
+		var/datum/ore_cloud_data/OCD = ores[material_name]
+		if(OCD)
+			amount_ejected = min(ejectamt, OCD.amount)
+			for(var/i in 1 to amount_ejected)
+				var/obj/item/raw_material/ore = new OCD.type_path(src)
```

Comment:
What's the reasoning for moving from storing the ore items directly to recreating them each time? Seems like it's making more work in future if we ever want to preserve more properties of the ore than what's listed here.

Reply:
Mostly because of `.amount` and its cursed nature. But also because some fuckery with ores being qdel'd and staying inside the rockbox and this seemed a more elegant solution.